core/src/main/java/software/amazon/awssdk/async/FileAsyncResponseHandler.java

```diff
@@ -121,7 +130,21 @@ public void onError(Throwable t) {
 
         @Override
         public void onComplete() {
+            // if write in progress, tell write to close on finish
+            synchronized (this) {
+                if (writeInProgress) {
+                    closeOnLastWrite = true;
+                } else {
+                    close();
+                }
+            }
+        }
+
+        private void close() {
             // Completion handled by response handler
```

Comment:
No longer true?

Reply:
Looks like this got left it while I was resolving some merge conflicts